### ISSUE TITLE
OSGi manifest headers missing in 3.22.0/3.22.1/3.22.2 [Java]

### ISSUE THREAD

**Issue by sjaenick** [protocolbuffers/protobuf#12017]
Hi there,

I just attempted to upgrade `protobuf-java` from 3.21.12 to 3.22.0 via maven, and noticed that
3.22.0 no longer contains OSGi manifest headers. Could this be fixed?

New dependency is:

```
<dependency>
            <groupId>com.google.protobuf</groupId>
            <artifactId>protobuf-java</artifactId>
            <version>3.22.0</version>
</dependency>
```

Thanks in advance.

**Comment by deannagarcia** [comment]
Thanks for this report! Working on a fix now and it should be out in 22.1 this week or next week.

**Comment by deannagarcia** [comment]
Sorry you're still hitting this issue, I think https://github.com/protocolbuffers/protobuf/issues/12170 might be the root cause. I'm working on that now.

**Comment by sjaenick** [comment]
Sorry, but stiill no luck. Maybe add a regression test to make sure `META-INF/MANIFEST.MF` has appropriate `Export-Package` statements?

**Comment by deannagarcia** [comment]
We are unfortunately aware that this didn't get fixed in 22.2. As linked, we are working on a fix to this now and will keep you updated. In the meantime it's probably best to use 21.12.

**Comment by googleberg** [comment]
@sjaenick I have a fix out for review. We don't have any OSGi experts on our team, would you be interested in contributing a test that would actually make sure that we don't break this in the future?

**Comment by sjaenick** [comment]
> @sjaenick I have a fix out for review. We don't have any OSGi experts on our team, would you be interested in contributing a test that would actually make sure that we don't break this in the future?

Could you take a look at https://github.com/protocolbuffers/protobuf/pull/12235 ?

**Comment by deannagarcia** [comment]
Closing this since the fix is in, we will be releasing this in 3.22.3 in the next few weeks.

**Comment by deannagarcia** [comment]
Update: I'm releasing 3.22.3 without this fix, but it will be in 23.0 which should be out next week.
